source/extensions/tracers/datadog/time_util.cc

```diff
@@ -0,0 +1,26 @@
+#include "source/extensions/tracers/datadog/time_util.h"
+
+#include <chrono>
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+
+datadog::tracing::TimePoint estimateTime(SystemTime wall) {
+  return estimateTime(wall, datadog::tracing::default_clock);
+}
+
+datadog::tracing::TimePoint estimateTime(SystemTime wall, const datadog::tracing::Clock& clock) {
+  datadog::tracing::TimePoint point = clock();
+  if (point.wall > wall) {
+    point.tick -= point.wall - wall;
```

Comment:
should we also verify that the delta we are subtracting is less than point.tick?

Reply:
`point.tick` is a time point, while `point.wall - wall` is a duration, so no.

Reply:
got it, thanks!